File: src/main/java/edu/fiuba/reservations/utils/validator/CompositeValidator.java
package edu.fiuba.reservations.utils.validator;

import edu.fiuba.reservations.utils.enums.Exception;
import java.util.ArrayList;
import java.util.List;

public final class CompositeValidator<T> implements Validator<T> {
    private final List<Validator<T>> validators;

    public CompositeValidator(List<Validator<T>> validators) {
        this.validators = validators;
    }

    public List<Exception> validate(T data, List<String> fieldNames) {
        ArrayList<Exception> exceptions = new ArrayList();

        validators.forEach(validator ->
            exceptions.addAll(validator.validate(data, fieldNames))
        );

        return exceptions;
    }
}

File: src/main/kotlin/edu/fiuba/reservations/utils/validator/CompositeValidator.kt
package edu.fiuba.reservations.utils.validator

import edu.fiuba.reservations.utils.enums.Exception

class CompositeValidator<T>(
    private val validators: List<Validator<T>>
) : Validator<T> {

    override fun validate(data: T, fieldNames: List<String>): List<Exception> {
        val exceptions = ArrayList<Exception>()

        validators.forEach { exceptions.addAll(it.validate(data, fieldNames)) }

        return exceptions
    }
}
